contrib/testrunner_pc/resources.c

```diff
@@ -0,0 +1,168 @@
+    if (fp == NULL) {
+        snprintf(path, MAX_PATH_LEN, "%s/%s", TEST_RESOURCE_DIR, name);
+        fp = fopen(path, "rb");
+        if (fp == NULL)
+            return NULL;
+    }
+    return fp;
+}
+
+static void prv_ensureResMappingCapacity() {
+    static const uint32_t BLOCK_SIZE = 16;
+    TestRunnerContext* context = &runner_context;
+    if (context->res_mapping.count == context->res_mapping.capacity) {
+        uint32_t new_capacity = context->res_mapping.capacity + BLOCK_SIZE;
+        
+        uint32_t* new_ids = (uint32_t*)realloc(context->res_mapping.ids, sizeof(uint32_t) * new_capacity);
+        if (new_ids == NULL) {
+            fputs("Could not allocate resource mapping ids", stderr);
+            exit(2);
+        }
+        const char** new_names = (const char**)realloc(context->res_mapping.names, sizeof(const char**) * new_capacity);
+        if (new_names == NULL) {
+            fputs("Could not allocate resource mapping names", stderr);
+            exit(2);
+        }
+
+        context->res_mapping.ids = new_ids;
+        context->res_mapping.names = new_names;
+        context->res_mapping.capacity = new_capacity;
+    }
+}
+
+bool int_ngfxtest_map_resource(const char* resource_name, uint32_t resource_id) {
+    TestRunnerContext* context = &runner_context;
+
+    // Try to open file
+    FILE* fp = openResourceByName(resource_name);
+    if (fp == NULL) {
+        return false;
+    }
+    fclose(fp);
+
+    // Search old entry
+    uint32_t i;
+    for (i = 0; i < context->res_mapping.count; i++) {
+        if (context->res_mapping.ids[i] == resource_id) {
+            context->res_mapping.names[i] = resource_name;
+            return true;
+        }
+    }
+
+    // Append new entry
+    prv_ensureResMappingCapacity();
+    context->res_mapping.ids[i] = resource_id;
+    context->res_mapping.names[i] = resource_name;
+    context->res_mapping.count++;
+    return true;
+}
+
+ResHandle resource_get_handle(uint32_t id) {
+    const char* name = getResourceNameById(id);
+    return (ResHandle)name;
+}
+
+size_t resource_size(ResHandle handle) {
+    if (handle == NULL) {
+        return 0;
+    }
+
+    FILE* fp = openResourceByName((const char*)handle);
+    if (fp == NULL) {
+        return 0;
+    }
+    fseek(fp, 0, SEEK_END);
+    size_t size = (size_t)ftell(fp);
+    fclose(fp);
+
+    return size;
+}
+
+size_t resource_load(ResHandle handle, uint8_t * buffer, size_t max_length) {
+    if (handle == NULL || buffer == NULL || max_length == 0) {
+        return 0;
+    }
+
+    FILE* fp = openResourceByName((const char*)handle);
+    if (fp == NULL) {
+        return 0;
+    }
+    size_t size_read = fread(buffer, 1, max_length, fp);
+    fclose(fp);
+
+    return size_read;
+}
+
+ResImage* loadImageById(uint32_t resource_id) {
+    const char* name = getResourceNameById(resource_id);
+    return name == NULL ? NULL : loadImageByName(name);
+}
+
+ResImage* loadImageByName(const char* name) {
```

Comment:
`ResImage *loadImageByName`, `const char *name`

Reply:
My colleagues don't call me "It's only a regex, right?" for nothing...